lib/node/runner/node_runner.go

```diff
@@ -682,25 +670,11 @@ func (nr *NodeRunner) NodeInfo() node.NodeInfo {
 	return nr.nodeInfo
 }
 
-func (nr *NodeRunner) startBroadcastBallot() {
-	for {
-		select {
-		case b := <-nr.ballotBroadcastChannel:
-			nr.broadcastBallot(b)
-		}
-	}
-}
-
 func (nr *NodeRunner) BroadcastBallot(b ballot.Ballot) {
 	go func() {
-		nr.ballotBroadcastChannel <- b
-	}()
-}
-
-func (nr *NodeRunner) broadcastBallot(b ballot.Ballot) {
-	go func() {
-		nr.handleBallotMessage(b)
+		encoded, _ := b.Serialize()
+		nr.Network().MessageBroker().Receive(common.NewNetworkMessage(common.BallotMessage, encoded))
 	}()
 
-	nr.ConnectionManager().Broadcast(b)
+	go nr.ConnectionManager().Broadcast(b)
```

Comment:
Should we call this as a go routine?
In this `Broadcast` function, all sending call is in go routine. And it has read lock.

Reply:
@Charleslee522 Yes, you right, it should not. I'll fix it.